### PR TITLE
Test262 results representation

### LINKED ISSUE #719: Use test262 data to generate page to show our conformance
Boa generates TEST262 data on every PR. 
It then dumps out some JSON files which chart how conformant we are compared to the test262 suite.

Currently we do nothing with this data.
It would be good to it to our `gh-pages` branch and use it to generate a static page.

You can test this locally by running.
`cargo run --release --bin boa_tester -- -o gh-pages/test262`
then inspecting the JSON output.

I'm easy on charting library.

1. We should have a new github action that runs test262 (weekly?) and generates data
2. Use that data to generate a page on github.io

### PR DESCRIPTION
**Pull request by Razican**
This PR adds a visual representation to the ECMAScript test suite results, it closes #719.

I also added "Prettier" formatting for all the custom files in the `gh-pages` branch.

### CODE REVIEW

**Comment by Razican** [comment]
A small peek on how this is looking:

<details>
<summary>Screenshot</summary>

</details>

Things I consider we should add:
 - version history on the right column (for now, only v0.10, but should add new ones automatically)
 - Progress bar for a visual representation on conformance
 - Graph to show the progress of the master branch.
 - passed/failed/ignored/panic information on test results (requires a change on how we generate results)
 - Links to test262 sources should be to the specific commit of the test262 repo (requires a change on how we generate results)
 - Loading spinning icons for when it's taking some time to load (full result files are huge, this could maybe be improved a bit by reducing the sizes of the JSON files with smaller object attributes.
 - Add a the name of the test on hover.
 - Maybe some style improvements.
 - Add links to benchmarks and conformance results from the homepage / demo.
 - Add links to information about test262 test suite and how to interpret the results.

Things already present:
 - Latest version and master branch overview
 - Fine-grained view of the working/failing tests
 - Click on a given test to directly go to the test262 repo, to the actual test file source
 - Formatted the code to use Prettier format.
 - Automatic updates on every commit to `master` and after every release.

File: test262/index.html [hunk 1]
```diff
@@ -0,0 +1,105 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <title>Boa EcmaScript conformance</title>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/png" href="../assets/favicon.png" />
+    <link href="../assets/bootstrap.min.css" rel="stylesheet" />
```
Comment (jasonwilliams) [inline test262/index.html:7]: These should link to the node_modules bootstrap, we shouldn’t be committing bootstrap files directly into he repo

**Review by jasonwilliams — CHANGES_REQUESTED** [review]

File: test262/index.html (hunk 1 above)
Comment (jasonwilliams) [reply test262/index.html:7]: Same with jquery, you may need webpack to link these dependencies together

Comment (Razican) [reply test262/index.html:7]: Hmmm we are already loading them for the benchmarks, and we don't use webpack for this. Maybe we should refactor everything in that regard.

Comment (jasonwilliams) [reply test262/index.html:7]: We decided to fix this up at a later date

**Comment by Razican** [comment]
I think this is ready for review. Some extra things can be added such as a graph for the master branch, but I think this is already pretty useful in the current form. I will open a couple of issues to:
 - Add a popover graph showing the progress of the `master` branch
 - Add the version history in the rightmost column
 - Rethink the gh-pages to not embed external libraries.

**Review by HalidOdat — APPROVED** [review]
Looks good to me! :)

Maybe we should add the link in `README.md`, for easier access

**Pull request merged**